File: 3163. String Compression III.py
class Solution:
    def compressedString(self, word: str) -> str:
        answer = ""
        i = 0
        while i < len(word):
            count = 1
            while count < 9 and i + 1 < len(word) and word[i] == word[i + 1]:
                i += 1
                count += 1
            answer += str(count) + word[i]
            i += 1
        return answer

File: 3163. String Compression III.java
class Solution {
    public String compressedString(String word) {
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < word.length(); ) {
            char c = word.charAt(i);
            int count = 0;
            while (i < word.length() && word.charAt(i) == c && count < 9) {
                count++;
                i++;
            }
            sb.append(String.valueOf(count));
            sb.append(c);
        }
        return sb.toString();
    }
}